### PR TITLE
fix: working version of RTK heading

### PR DESCRIPTION
**Pull request by doudou**
I could verify that the heading matches the one from the logs, and that the driver would report it. However, the GPS
logs did not match the gyro behaviour on my table, which created nonsensical dynamic behaviour - expected.

I believe there are quite a few issues to solve still, but this could be workable if we increase the accuracy.

### CODE REVIEW

File: lib/Core/GPS/src/processUbloxGPS.c [hunk 1]
```diff
@@ -499,15 +499,15 @@ void decodeNavPvt(char *msg, GpsData_t *GPSData)
 }
 
 void extractHeadingFromRelPosNed(char *msg, GpsData_t* GPSData) {
-    int relPosHeading = *( (int *) msg + 24 );
+    int32_t relPosHeading = *(int32_t*)(msg + 24);
```
Comment (wvmcastro) [inline lib/Core/GPS/src/processUbloxGPS.c:502]: awf, I realize now that when you pointed that back in the office I looked the precedence table wrongly ... I saw unary operator +/- when I should have looked for addition operator. But even if it wasn't the case, I agree that better parentization would led to more readable and a lot less error prone code.

**Review by wvmcastro — APPROVED** [review]

**Comment by wvmcastro** [comment]
I would like to add unit tests for this repo, I think some of the bugs could have been caught earlier (the relposned parsing one). I tried to setup an unit test suite but without success

File: lib/Core/GPS/src/processUbloxGPS.c (hunk 1 above)
Comment (doudou) [reply lib/Core/GPS/src/processUbloxGPS.c:502]: Well. You know

IF YOU NEED TO READ A TABLE TO THINK THAT YOU KNOW PRECEDENCE, DON'T AND USE PARENTHESIS.

**Comment by doudou** [comment]
> I would like to add unit tests for this repo, I think some of the bugs could have been caught earlier (the relposned parsing one). I tried to setup an unit test suite but without success

I think it's very low priority. We don't plan to develop this firmware a lot, and it would be a lot of work.

A.k.a. please leave this IMU firmware alone until you come back to the office and focus on the particle filter. Please.

**Pull request merged**